### DISCUSSION TITLE
Is there a way to map the results(constraint status) of VerificationSuite's run() method to the metrics stored in metrics repository?

### DISCUSSION THREAD

**Discussion by sharadcapillary** [awslabs/python-deequ#141] [Q&A]
Hi, I was building a system over deequ, where ppl can write test cases on data using deequ and save the run metrics for visualization and later anomaly detection.  However the metrics that get saved in the metrics repository by default don't contain the constraint status i.e., whether a constraint passed or failed. It just holds the value. I want my system to be automated, and I want to store the metrics and the run status for each constraint. 

1. So is there any way to map the result of the .run() command to each row in the metrics stored in the metrics repository by deequ? 
2. Also Is there a way to save different Result Key w.r.t each constraint for the same "Check" object?

I am aware that I can achieve the above using different "Check" objects for each constraint, however that would be very inefficient I believe.

Image shows the result of VerificationSuite.run()

<img width="1339" alt="image" src="https://github.com/awslabs/python-deequ/assets/134647394/0795e314-880f-4bda-8693-3152a9ca4bb1">

Image shows the result of loading the metrics from metrics repository

<img width="582" alt="image" src="https://github.com/awslabs/python-deequ/assets/134647394/48bcef3a-c178-4c6e-ac4f-3ec8128f48e1">

**Comment by dwalter469** [comment]
Follow on to that question, is there a way to save the Check Level (warning, error, information) to the MetricRepository along withe the Constraint Status. Having this information in the Metric Repository would be very helpful.
